src/Microsoft.Health.Dicom.Functions/Management/PurgeOrchestrationInstancesHistory.cs

```diff
@@ -0,0 +1,88 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Health.Dicom.Functions.Management
+{
+    public class PurgeOrchestrationInstancesHistory
+    {
+        private readonly OrchestrationsHistoryConfiguration _purgeConfig;
+        private readonly Func<DateTime> _getUtcNow;
+
+        public PurgeOrchestrationInstancesHistory(IOptions<OrchestrationsHistoryConfiguration> cleanupOptions)
+            : this(cleanupOptions, () => DateTime.UtcNow)
+        { }
+
+        internal PurgeOrchestrationInstancesHistory(IOptions<OrchestrationsHistoryConfiguration> cleanupOptions, Func<DateTime> getDateTimeUtcNow)
+        {
+            EnsureArg.IsNotNull(cleanupOptions?.Value, nameof(cleanupOptions));
+            EnsureArg.IsNotNull(getDateTimeUtcNow, nameof(getDateTimeUtcNow));
+            _purgeConfig = cleanupOptions?.Value;
```

Comment:
nit: you don't need the [`null`-conditional operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-) (`?.`) as you `null`-checked above